File: 13rd_Competition/Jongryul/03/16395.js
const fs = require("fs");
const filePath = process.platform === "linux" ? "/dev/stdin" : "input.txt";
const [n, k] = fs
  .readFileSync(filePath)
  .toString()
  .trim()
  .split(" ")
  .map(Number);
const dp = Array.from({ length: n }, (_, i) => Array(i + 1).fill(0));

for (let i = 0; i < n; i++) {
  for (let j = 0; j < i + 1; j++) {
    if (!j || j === i) {
      dp[i][j] = 1;
    } else {
      dp[i][j] = dp[i - 1][j - 1] + dp[i - 1][j];
    }
  }
}

console.log(dp[n - 1][k - 1]);

File: 01_Season/13rd_Competition/Jongryul/03/16395.py
n, k = map(int, input().split())
dp = [[0] * i for i in range(1, n+1)]

for i in range(n):
	for j in range(i+1):
		if j == i or not j:
			dp[i][j] = 1
		else:
			dp[i][j] = dp[i-1][j-1] + dp[i-1][j]
print(dp[n-1][k-1])